packages/components/src/dropdown/index.tsx

```diff
@@ -42,12 +47,14 @@ export default function Dropdown( props ) {
 		onClose,
 		onToggle,
 		style,
-	} = props;
+	}: WordPressComponentProps< DropdownProps, 'div' >,
```

Comment:
The `WordPressComponentProps< SomeProps, 'div' >` construct will allow _all_ props that a `div` can take, so it basically only makes sense when we're passing the `...restProps` to the div.

Since were not passing rest props here, I think it would be better to not use `WordPressComponentProps`, and instead explicitly add types for `className` and `style`.

```suggestion
	}: DropdownProps,
```

Reply:
Good point, this doesn't accept any `WordPressComponentProps`, only `className` and `style.`

Committed in https://github.com/WordPress/gutenberg/compare/1f9ff853fa72a2a784be08449b7f99561bc647ad...181cfdae721e7c1dccf043fb19ec92cbab9f3a63